### PR TITLE
Improve folder selection in wizards

### PR DESCRIPTION
**Pull request by ckamm**
Primarily address #5597: Don't propose existing folders as sync folders in wizards to not cause accidents.

### CODE REVIEW

**Comment by SamuAlfageme** [comment]
Note to self; check what happens with the finder integration that creates a sync folder shortcut in the left bar when the default folder name was used.

**Comment by SamuAlfageme** [comment]
@ckamm also one of my wishes for a future complete folder selection _experience_ in the account wizard would be https://github.com/owncloud/client/issues/3632 😇

What do you think?

**Comment by ckamm** [comment]
@SamuAlfageme That is indeed occasionally confusing users, but would need a significant account creation wizard redesign. Currently the ability to only sync a remote subfolder is not being exposed in the folder setup part of the account creation wizard.

File: src/gui/folderman.h [hunk 1]
```diff
@@ -124,11 +124,24 @@ class FolderMan : public QObject
...
+    /**
+     * Attempts to find a non-existing, acceptable path for creating a new sync folder.
+     *
+     * Uses \a basePath as the baseline. It'll return this path if it's acceptable.
+     *
+     * Note that this can fail. If someone syncs ~ and \a basePath is ~/ownCloud, no
+     * subfolder of ~ would be a good candidate. When that happens \a ok will be
+     * false and \a basePath is returned.
+     */
+    QString findGoodPathForNewSyncFolder(const QString &basePath, const QUrl &serverUrl, bool *ok = 0) const;
```
Comment (ogoffart) [inline src/gui/folderman.h]: if ok is not used, maybe we should leave it out.

Comment (ckamm) [inline src/gui/folderman.h]: True

**Comment by ogoffart** [comment]
:+1:  
Maybe write some tests.

**Comment by ogoffart** [comment]
:+1:

**Pull request merged**